src/secrets/CommandMkdir.ts

```diff
@@ -50,16 +54,41 @@ class CommandMkdir extends CommandPolykey {
           },
           logger: this.logger.getChild(PolykeyClient.name),
         });
-        await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsMkdir({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              dirName: secretPath[1],
-              recursive: options.recursive,
-            }),
-          meta,
-        );
+        await binUtils.retryAuthentication(async (auth) => {
+          const response =
+            await pkClient.rpcClient.methods.vaultsSecretsMkdir();
+          const writer = response.writable.getWriter();
+          let first = true;
+          for (const [vault, path] of secretPaths) {
+            await writer.write({
+              nameOrId: vault,
+              dirName: path,
+              metadata: first
+                ? { ...auth, options: { recursive: options.recursive } }
+                : undefined,
+            });
+          }
+          await writer.close();
+          for await (const result of response.readable) {
+            if (result.type === 'error') {
+              const error = result as ErrorMessage;
+              let message: string = '';
+              if (error.code === 'ENOENT') {
+                message = 'No such secret or directory';
+              } else if (error.code === 'EEXIST') {
+                message = 'Secret or directory exists';
+              } else {
+                throw new ErrorPolykeyCLIUncaughtException(
+                  `Unexpected error code ${error.code}`,
+                );
```

Comment:
I need to check if the incoming error is not what we expected, even if the two possible codes are `ENOENT` and `EEXIST`. However, I'm not sure what is the best way to check that and confirm the error codes are valid.

Currently, I am throwing an `ErrorPolykeyCLIUncaughtException`, but that doesn't seem as if it should be thrown intentionally. Should I not worry about invalid error codes? Or is my current approach to check for valid error code is fine?

Reply:
When checking a single value like this you should use a `switch() case` block. It's much cleaner.

If  the default case is expected to never happen then always use a `utils.never('why this shouldn't happen')` as an invalid logic error.

```
switch(error.code){
  case 'ENOENT': message = 'whatever'
  break;
  case 'EEXIST: message = 'whatever'
  break;
  default:
    utils.never('some message');
}
```